nucypher/network/middleware.py

```diff
@@ -115,6 +118,16 @@ def method_wrapper(path,
             else:
                 certificate_filepath = node_certificate_filepath
 
+            if node_or_sprout:
+                parsed = urllib.parse.urlparse(host)
+                try:
+                    certificate = ssl.get_server_certificate(addr=(parsed.hostname, parsed.port))
+                    certificate_filepath = node_or_sprout.node_storage.store_node_certificate(certificate)
+                    self.log.warn(f"Replaced SSL certificate for host: {host}")
+                except:
+                    self.log.warn(f"Failed to replace SSL certificate for host: {host}")
+                    pass
```

Comment:
What exceptions are you anticipating here?  Do you think it will be okay to suppress *all* of them?

Reply:
Just to clarify, this PR is not in its intended state, and the exceptions should be handled on a case-to-case basis. I'm probably going to take a different approach in solving this issue.